tools/generator/src/Generator+AddBazelDependenciesTarget.swift

```diff
@@ -0,0 +1,398 @@
+        .resolve(.internal(rsyncFileListPath), useScriptVariables: true)
+        .string
+)" \
+  --chmod=u+w \
+  -L \
+  . \
+  "$BUILD_DIR/bazel-out"
+
+"""#,
+            showEnvVarsInLog: false
+        )
+        pbxProj.add(object: script)
+
+        return script
+    }
+
+    private static func createFixModulemapsScript(
+        in pbxProj: PBXProj,
+        files: [FilePath: File],
+        filePathResolver: FilePathResolver
+    ) -> PBXShellScriptBuildPhase? {
+        guard files.containsModulemaps else {
+            return nil
+        }
+
+        let script = PBXShellScriptBuildPhase(
+            name: "Fix Modulemaps",
+            inputFileListPaths: [
+                filePathResolver
+                    .resolve(.internal(modulemapsFileListPath))
+                    .string,
+            ],
+            outputFileListPaths: [
+                filePathResolver
+                    .resolve(.internal(fixedModulemapsFileListPath))
+                    .string,
+            ],
+            shellScript: #"""
+set -eu
+
+while IFS= read -r input; do
+  output="${input%.modulemap}.xcode.modulemap"
+  perl -p -e \
+    's%^(\s*(\w+ )?header )(?!("\.\.(\/\.\.)*\/|")(bazel-out|external)\/)("(\.\.\/)*)(.*")%\1\6SRCROOT/\8%' \
+    < "$input" \
+    > "$output"
+done < "$SCRIPT_INPUT_FILE_LIST_0"
+
+"""#,
+            showEnvVarsInLog: false
+        )
+        pbxProj.add(object: script)
+
+        return script
+    }
+
+    private static func createFixInfoPlistsScript(
+        in pbxProj: PBXProj,
+        files: [FilePath: File],
+        filePathResolver: FilePathResolver
+    ) -> PBXShellScriptBuildPhase? {
+        guard files.containsInfoPlists else {
+            return nil
+        }
+
+        let script = PBXShellScriptBuildPhase(
+            name: "Fix Info.plists",
+            inputFileListPaths: [
+                filePathResolver
+                    .resolve(.internal(infoPlistsFileListPath))
+                    .string,
+            ],
+            outputFileListPaths: [
+                filePathResolver
+                    .resolve(.internal(fixedInfoPlistsFileListPath))
+                    .string,
+            ],
+            shellScript: #"""
+set -eu
+
+while IFS= read -r input; do
+  output="${input%.plist}.xcode.plist"
+  cp "$input" "$output"
+  plutil -remove UIDeviceFamily "$output" || true
+done < "$SCRIPT_INPUT_FILE_LIST_0"
+
+"""#,
+            showEnvVarsInLog: false
+        )
+        pbxProj.add(object: script)
+
+        return script
+    }
+}
+
+private extension Dictionary where Key == FilePath {
+    var containsModulemaps: Bool {
+        contains(where: { filePath, _ in
+            return filePath.type == .generated
+                && filePath.path.extension == "modulemap"
+        })
+    }
+
+    var containsInfoPlists: Bool {
+        contains(where: { filePath, _ in
+            return filePath.type == .generated
+                && filePath.path.lastComponent == "Info.plist"
```

Comment:
Are info plists guaranteed to have this exact file name? Isn't it possible to specify an info plist with a different name? If that's the case, maybe a comment to that effect could be helpful.

Since this file is generated, maybe we control that and it's not a problem?

Reply:
They are generated with this name in rules_apple. Also, when #293 is tackled we won't depend on this heuristic anymore.